net/ddns-scripts/files/usr/lib/ddns/update_porkbun_v3.sh

```diff
@@ -0,0 +1,162 @@
+#
+# Distributed under the terms of the GNU General Public License (GPL) version 2.0
+# 2024 Ansel Horn <dev@cahorn.net>
+#
+# Script for DDNS support via Porkbun's v3 API for the OpenWRT ddns-scripts package.
+#
+# Will attempt to create a new or edit an existing A or AAAA record for the
+# given domain and subdomain. Existing CNAME and ALIAS records WILL NOT BE
+# EDITED OR DELETED!  "username" and "password" configurations should be set to
+# Porkbun API key and secret key, respectively.
+#
+# Porkbun API documentation:
+# https://porkbun.com/api/json/v3/documentation#DNS%20Create%20Record
+#
+
+# Source JSON parser
+. /usr/share/libubox/jshn.sh
+
+# Set API base URL
+# Porkbun has warned it may change API hostname in the future:
+# https://porkbun.com/api/json/v3/documentation#apiHost
+__API="https://api.porkbun.com/api/json/v3"
+
+# Check availability of cURL with SSL
+[ -z "$CURL" ] && [ -z "$CURL_SSL" ] && write_log 14 "cURL with SSL support required! Please install"
+
+# Validate configuration
+[ -z "$domain" ] && write_log 14 "Service section not configured correctly! Missing 'domain'"
+[ -z "$username" ] && write_log 14 "Service section not configured correctly! Missing 'username'"
+[ -z "$password" ] && write_log 14 "Service section not configured correctly! Missing 'password'"
+
+# Split FQDN into domain and subdomain(s)
+__DOMAIN_REGEX='^\(\(.*\)\.\)\?\([^.]\+\.[^.]\+\)$'
+echo $domain | grep "$__DOMAIN_REGEX" > /dev/null || write_log 14 "Invalid domain! Check 'domain' config"
+__DOMAIN=$(echo $domain | sed -e "s/$__DOMAIN_REGEX/\3/")
+__SUBDOMAIN=$(echo $domain | sed -e "s/$__DOMAIN_REGEX/\2/")
+
+# Determine IPv4 or IPv6 address and record type
+if [ "$use_ipv6" -eq 1 ]; then
+	expand_ipv6 "$__IP" __ADDR
+	__TYPE="AAAA"
+else
+	__ADDR="$__IP"
+	__TYPE="A"
+fi
+
+
+# Inject authentication into API request JSON payload
+function json_authenticate() {
+	json_add_string "apikey" "$username"
+	json_add_string "secretapikey" "$password"
+}
+
+# Make Porkbun API call
+# $1 - Porkbun API endpoint
+# $2 - request JSON payload
+function api_call() {
+	local response url
+	url="$__API/$1"
+	write_log 7 "API endpoint URL: $url"
+	write_log 7 "API request JSON payload: $2"
+	response=$($CURL --data "$2" "$url")
+	write_log 7 "API response JSON payload: $response"
+	echo "$response"
+
```

Comment:
```suggestion
}
```

Reply:
I do not have this ddns provider, so I couldn't test it. That slipped through my fingers during the check! Thanks for spotting this issue.
Fixed with https://github.com/openwrt/packages/commit/d9cc0e1cd42833f1e6adee7d7abf9b4101b9c1fb

Reply:
The write_log calls in api_call seem to mess up parsing when I run `dynamic_dns_updater.sh -v 2 start`. If I remove the write_log lines in api_call, it runs fine. I think the write_log gets redirected to stdout which then get captured as part of the return value for api_call when run with -v 2. Removing the write_log fixes the issue.

@ownaginatious or @feckert , can you confirm if this makes sense.

Reply:
Since the `ddns-scripts` packages is technically quite messed up. I think the easiest fix is to remove these log messages.
@slowsage would you please open a PR?
@cahorn Or would you like to look at it again to see if you can find a better solution?